pkg/provider/kubernetes/crd/kubernetes.go

```diff
@@ -278,7 +278,9 @@ func loadServers(client Client, namespace string, svc v1alpha1.Service) ([]confi
 			}
 
 			protocol := "http"
-			if port == 443 || strings.HasPrefix(portSpec.Name, "https") {
+			if svc.Scheme != "" {
```

Comment:
```suggestion
						if port.Port == 443 || strings.HasPrefix(portSpec.Name, "https") {
							protocol = "https"
						}

						switch svc.Scheme {
						case "https":
						case "h2c":
						case "http":
						default:
							log.Errorf("Invalid scheme %s specified , svc.Scheme)
							continue
						}

```

Reply:
This prevents people from setting the scheme to be:
`http://mysecretendpoint.com;` and hijacking requests ;)